core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/Path.kt

```diff
@@ -41,6 +41,9 @@ interface Path {
 
     @JvmName("getTrackLocationAtOffset")
     fun getTrackLocationAtOffset(pathOffset: Distance): TrackLocation
+
+    @JvmName("getTrackSections")
+    fun getTrackSections(): List<TrackLocation>
```

Comment:
`TrackLocation` generally refers to a track and a location on this track, it's not really meant to be used that way.

There's `IdxWithOffset` for this purpose